README.md

```diff
@@ -62,33 +61,117 @@ Here are the valid sampling states. Note they all applied to the trace ID, not t
 
 The most common use of sampling is probablistic: eg, accept 0.01% of traces and deny the rest. Debug is the least common use case.
 
-# Http Encoding
-B3 attributes are most commonly propagated as Http headers. All B3 headers follows the convention of `X-B3-${name}` with special-casing for flags. When reading headers, the first value wins.
+# Http Encodings
+There are two encodings of B3: Single Header and Multiple Header. Multiple header encoding uses an `X-B3-` prefixed header per item in the trace context. Single header delimits the context into into a single entry named `b3`. When both are present, the single-header variant is preferred.
```

Comment:
Not sure if it is clear to the uninitiated what "the single-header variant is preferred" means. Preferred is a bit fuzzy and it lacks explanation on this being an extraction detail (as we can't be clear enough about scope).

"the single-header variant takes precedence over the multiple header one when extracting the details"

Reply:
much better